src/pages/PaymentDetails.tsx

```diff
@@ -59,6 +78,38 @@ export const PaymentDetails = () => {
     navigate(`${Routes.RECEIVERS}/${receiverId}`);
   };
 
+  const showModal = (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    event.preventDefault();
+    setIsModalVisible(true);
+  };
+
+  const hideModal = (
+    event?: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    event?.preventDefault();
+    setIsModalVisible(false);
+  };
+
+  const handleCancelPayment = (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    event.preventDefault();
+    cancelPayment({ paymentId: formattedPayment?.id });
```

Comment:
I think it's pretty safe to assume that we will always have payment ID here (otherwise the cancel button wouldn't be rendered). We can add a check here to make TypeScript happy. What do you think? :nerd_face: